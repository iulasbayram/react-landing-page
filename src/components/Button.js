import React, { PureComponent } from 'react';
import { css, style } from 'glamor';

import Globals from '../utils/Globals';

const cont = css({
  textAlign: 'center',
  textTransform: 'uppercase',
  borderRadius: 4,
  padding: '8px 40px',
  color: Globals.colors.darken,
  fontWeight: 'bold',
  cursor: 'pointer',
  border: `1px solid ${Globals.colors.primary}`,
  '> h2': { margin: 0, letterSpacing: 2 },
  backgroundColor: Globals.colors.primary,
  transition: Globals.transitions.primary,

  ':hover': {
    backgroundColor: 'transparent',
    color: Globals.colors.primary,
  },

  '@media(max-width: 720px)': {
    margin: '5px',
  },
});

const background = css({
  padding: 10,
  backgroundRepeat: 'no-repeat',
  backgroundPosition: 'center',
  backgroundSize: 'cover',
  margin: '30px',
});

const Input = css({
  background: 'transparent',
  fontSize: '1.5em',
});

export default class Button extends PureComponent {
  state = { active: false };

  toggleActive = () => {
    this.setState(prevState => ({
      active: !prevState.active,
    }));
  };

  render() {
    const { label, pattern, newsletter } = this.props;
    const { active } = this.state;

    return newsletter
      ? <div
          className={background}
          {...style({ ':hover': { backgroundImage: `url('${pattern}')` } })}
        >
          <div className={cont} onClick={this.toggleActive}>
            {active
              ? <input
                  type="email"
                  className={Input}
                  placeholder="Seu email aqui"
                />
              : <h2>
                  {' '}{label}{' '}
                </h2>}
          </div>
        </div>
      : <div
          className={background}
          {...style({ ':hover': { backgroundImage: `url('${pattern}')` } })}
        >
          <div className={cont}>
            <h2>
              {label}
            </h2>
          </div>
        </div>;
  }
}
