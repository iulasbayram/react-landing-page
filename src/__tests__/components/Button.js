import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import TestUtils from 'react-addons-test-utils';
import Button from '../../components/Button';

describe('<Button />', () => {
  it('render self', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.is('div')).toBe(true);
  });

  it('should render a label', () => {
    const wrapper = shallow(<Button label="MyLabel" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a button label', () => {
    const wrapper = shallow(<Button label="MyLabel" />);
    expect(wrapper.prop('label')).not.toEqual('MyLabel');
  });

  // it('should render a document title and a parent title', () => {
  //   const wrapper = shallow(
  //     <Button label="MyLabel" pattern="pattern" />
  //   );
  //   expect(wrapper.prop('label')).toEqual('MyLabel â€” pattern');
  // });
});
