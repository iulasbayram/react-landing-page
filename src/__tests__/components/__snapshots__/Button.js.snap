exports[`<Button /> should render a label 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={
        Object {
            "data-css-1yrpvnn": "",
          }
    }
    data-css-a3nw7g="">
    <div
        className={
            Object {
                "data-css-1knz92t": "",
              }
        }>
        <h2>
            MyLabel
        </h2>
    </div>
</div>,
  "nodes": Array [
    <div
      className={
            Object {
                  "data-css-1yrpvnn": "",
                }
      }
      data-css-a3nw7g="">
      <div
            className={
                  Object {
                        "data-css-1knz92t": "",
                      }
            }>
            <h2>
                  MyLabel
            </h2>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Button
        label="MyLabel" />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "label": "MyLabel",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={
                    Object {
                              "data-css-1yrpvnn": "",
                            }
          }
          data-css-a3nw7g="">
          <div
                    className={
                              Object {
                                        "data-css-1knz92t": "",
                                      }
                    }>
                    <h2>
                              MyLabel
                    </h2>
          </div>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className={
                    Object {
                              "data-css-1yrpvnn": "",
                            }
          }
          data-css-a3nw7g="">
          <div
                    className={
                              Object {
                                        "data-css-1knz92t": "",
                                      }
                    }>
                    <h2>
                              MyLabel
                    </h2>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    label="MyLabel" />,
}
`;
